src/sentry/models/orgauthtoken.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils import timezone
+from django.utils.encoding import force_text
+
+from sentry.conf.server import SENTRY_SCOPES
+from sentry.db.models import (
+    ArrayField,
+    BaseManager,
+    FlexibleForeignKey,
+    Model,
+    control_silo_only_model,
+    sane_repr,
+)
+from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
+from sentry.models.project import Project
+
+
+def validate_scope_list(value):
+    for choice in value:
+        if choice not in SENTRY_SCOPES:
+            raise ValidationError(f"{choice} is not a valid scope.")
+
+
+@control_silo_only_model
+class OrgAuthToken(Model):
+    __include_in_export__ = True
+
+    organization_id = HybridCloudForeignKey("sentry.Organization", null=False, on_delete="CASCADE")
+    # The JWT token in hashed form
+    token_hashed = models.CharField(max_length=3000, unique=True, null=False)
```

Comment:
I recommend we use SHA512 to hash these tokens. That gives us 128 characters for the hashed portion + the expected prefix `sentrys_` (8 characters). So the `max_length` can be set to 136. 🙂

Reply:
I would skip defining the length on the field. Postgres doesn't do [any optimizations on char fields](https://wiki.postgresql.org/wiki/Don%27t_Do_This#Don.27t_use_varchar.28n.29_by_default) if they have fixed/static length. Changing the length requires taking an exclusive lock on the table which is a category of operation we try to minimize.

Reply:
Makes sense, I'll remove the `max_length` then!